### ISSUE TITLE
matchers: nicer syntax for Type testing

### ISSUE THREAD

**Issue by DartBot** [dart-lang/test#214]
<a href="https://github.com/seaneagan"><img src="https://avatars.githubusercontent.com/u/444270?v=3" align="left" width="96" height="96"hspace="10"></img></a> **Issue by [seaneagan](https://github.com/seaneagan)**
_Originally opened as dart-lang/sdk#10407_

----

**What steps will reproduce the problem?**
**1.**
**2.**
**3.**

**What is the expected output? What do you see instead?**

**What version of the product are you using? On what operating system?**

currently it doesn't read as nicely as other matchers:

expect(x, new isInstanceOf&lt;int&gt;());
expect(f, throwsA(new isInstanceOf&lt;ExpectedError&gt;()));

better would be:

expect(x, isA(int));
expect(f, throwsA(isA(ExpectedError)));

or even:

expect(x, int);
expect(f, throwsA(ExpectedError));

I think this is blocked on issue dart-lang/sdk#10406.

**Please provide any additional information below.**

**Comment by DartBot** [comment]
<a href="https://github.com/madsager"><img src="https://avatars.githubusercontent.com/u/2909286?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [madsager](https://github.com/madsager)**

----

 cc @sigmundch.
*Added [Area-Pkg](../labels/Area-Pkg), [Triaged](../labels/Triaged) labels.*

**Comment by DartBot** [comment]
<a href="https://github.com/sigmundch"><img src="https://avatars.githubusercontent.com/u/2049220?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sigmundch](https://github.com/sigmundch)**

----

 cc @gramster.
*Removed [Area-Pkg](../labels/Area-Pkg) label.*
*Added [Area-unittest](../labels/Area-unittest) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/gramster"><img src="https://avatars.githubusercontent.com/u/2762632?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [gramster](https://github.com/gramster)**

----

Initially we took this approach because we could not rely on 'is' in dart2js if the type was passed as a parameter. Perhaps there is enough reflection support now to make this work; will have to investigate.

In the meantime, a workaround is to define matcher instances for the types you care about (the library already does this for most exception and error types). E.g. 

final is_int = new predicate((x) => x is int, "is an int");

**Comment by DartBot** [comment]
<a href="https://github.com/gramster"><img src="https://avatars.githubusercontent.com/u/2762632?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [gramster](https://github.com/gramster)**

----

*Added [Duplicate](../labels/Duplicate) label.*
*Marked as being merged into dart-lang/sdk#8138.*

**Comment by DartBot** [comment]
<a href="https://github.com/kevmoo"><img src="https://avatars.githubusercontent.com/u/17034?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [kevmoo](https://github.com/kevmoo)**

----

*Added [Pkg-Unittest](../labels/Pkg-Unittest) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/kevmoo"><img src="https://avatars.githubusercontent.com/u/17034?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [kevmoo](https://github.com/kevmoo)**

----

*Removed [Area-unittest](../labels/Area-unittest) label.*
*Added [Area-Pkg](../labels/Area-Pkg) label.*
